spec/models/miq_server/worker_management/monitor_spec.rb

```diff
@@ -33,6 +33,20 @@
 
     context "#sync_workers" do
       let(:server) { EvmSpecHelper.local_miq_server }
+
+      it "ensures only expected worker classes are constantized" do
+        # Autoload abstract base class for the event catcher
+        ManageIQ::Providers::BaseManager::EventCatcher
+
+        # We'll try to constantize a non-existing EventCatcher class in an existing namespace,
+        # which incorrectly resolves to the base manager event catcher.
+        allow(MiqServer).to receive(:monitor_class_names).and_return(%w(ManageIQ::Providers::Foreman::ProvisioningManager::EventCatcher MiqGenericWorker))
+
+        expect(ManageIQ::Providers::BaseManager::EventCatcher).to receive(:sync_workers).never
```

Comment:
Is `expect(ManageIQ::Providers::BaseManager::EventCatcher).not_to receive(:sync_workers)` functionally equivalent?  I think it reads better

Reply:
Yes, I agree